### DISCUSSION TITLE
symfony/notifier with Microsoft Teams

### DISCUSSION THREAD

**Discussion by CyrilCharlier** [symfony/symfony#49066] [Q&A] [answered]
Hello,

I try to use the official notifier with Microsoft Teams : 

https://symfony.com/doc/current/notifier.html

this is my "notifier.yaml" :  (i don't found a sample with Teams)

``` yaml
framework:
    notifier:
        chatter_transports:
            microsoftteams: '%env(DSN_TEAMS)%'
```

and my Controller for test :

``` php
<?php

namespace App\Controller;

use Symfony\Bundle\FrameworkBundle\Controller\AbstractController;
use Symfony\Component\HttpFoundation\Response;
use Symfony\Component\Routing\Annotation\Route;
use Symfony\Component\Notifier\Notification\Notification;
use Symfony\Component\Notifier\NotifierInterface;
use Symfony\Component\Notifier\Recipient\NoRecipient;

class DashboardController extends AbstractController
{
    #[Route('/dashboard', name: 'app_dashboard')]
    public function index(NotifierInterface $notifier): Response
    {
        $notification = (new Notification('Test Cyril', ['microsoftteams']))
            ->content('Test Chatter In Channel');
        
        $recipient = new NoRecipient();

        $notifier->send($notification, $recipient);

        return $this->render('dashboard.html.twig', [
            'controller_name' => 'DashboardController',
        ]);
    }
}
```

and this is the result : 

<img width="800" alt="image" src="https://user-images.githubusercontent.com/11455745/213925978-873b319d-4a6b-4574-a601-d5e84b387783.png">

I can't find a solution for that, any idea ?

And of cours i have installed the good package : 

``` json
        "symfony/microsoft-teams-notifier": "6.2.*",
        "symfony/notifier": "6.2.*",
```
thanks !

**Comment by MatTheCat** [comment, marked as answer]
Hello,

it seems you’re confusing **channels** with **transports**. Here, `symfony/microsoft-teams-notifier` provides a transport for the `chat` channel, so your notification should be

```php
$notification = (new Notification('Test Cyril', ['chat']))
            ->content('Test Chatter In Channel');
```

Note that if you know you’ll always use this transport you can use the `Chatter` rather than the `Notifier`: https://symfony.com/doc/current/notifier/teams.html

**Reply by CyrilCharlier** [reply]
Thanks for the reply. It is ok no more error. 
I will have a look at the link you give me !
Thanks
